lib/Backend/BackwardPass.cpp

```diff
@@ -252,6 +252,84 @@ BackwardPass::CleanupBackwardPassInfoInFlowGraph()
 }
+
+/*
+*   We Insert ArgIns at the start of the function for all the formals.
+*   Unused formals will be deadstored during the deadstore pass.
+*   We need ArgIns only for the outermost function(inliner).
+*/
+void
+BackwardPass::InsertArgInsForFormals()
+{
+    if (func->IsStackArgsEnabled() && !func->GetJnFunction()->GetHasImplicitArgIns())
+    {
+        IR::Instr * insertAfterInstr = func->m_headInstr->m_next;
+        AssertMsg(insertAfterInstr->IsLabelInstr(), "First Instr of the first block should always have a label");
+
+        Js::ArgSlot paramsCount = insertAfterInstr->m_func->GetJnFunction()->GetInParamsCount() - 1;
+        IR::Instr *     argInInstr = nullptr;
+        for (Js::ArgSlot argumentIndex = 1; argumentIndex <= paramsCount; argumentIndex++)
+        {
+            IR::SymOpnd *   srcOpnd;
+            StackSym *      symSrc = StackSym::NewParamSlotSym(argumentIndex + 1, func);
+            StackSym *      symDst = StackSym::New(func);
+            IR::RegOpnd * dstOpnd = IR::RegOpnd::New(symDst, TyVar, func);
+
+            func->SetArgOffset(symSrc, (argumentIndex + LowererMD::GetFormalParamOffset()) * MachPtr);
+
+            srcOpnd = IR::SymOpnd::New(symSrc, TyVar, func);
+
+            argInInstr = IR::Instr::New(Js::OpCode::ArgIn_A, dstOpnd, srcOpnd, func);
+            insertAfterInstr->InsertAfter(argInInstr);
+            insertAfterInstr = argInInstr;
+
+            AssertMsg(!func->HasStackSymForFormal(argumentIndex - 1), "Already has a stack sym for this formal?");
+            this->func->TrackStackSymForFormalIndex(argumentIndex - 1, symDst);
+        }
+
+        if (PHASE_VERBOSE_TRACE1(Js::StackArgFormalsOptPhase))
+        {
+            Output::Print(_u("StackArgFormals : %s (%d) :Inserting ArgIn_A for LdSlot (formals) in the start of Deadstore pass. \n"), func->GetJnFunction()->GetDisplayName(), func->GetJnFunction()->GetFunctionNumber());
+            Output::Flush();
+        }
+    }
+}
+
+void
+BackwardPass::ProcessBailOnStackArgsOutOfActualsRange()
+{
+    IR::Instr * instr = this->currentInstr;
+
+    if (tag == Js::DeadStorePhase && instr->m_opcode == Js::OpCode::LdElemI_A && instr->HasBailOutInfo() && !IsPrePass())
+    {
+        if (instr->DoStackArgsOpt(this->func))
+        {
+            AssertMsg(instr->GetBailOutKind() & IR::BailOnStackArgsOutOfActualsRange, "Stack args bail out is not set when the optimization is turned on? ");
+            if (instr->GetBailOutKind() & ~IR::BailOnStackArgsOutOfActualsRange)
+            {
+                Assert(instr->GetBailOutKind() == (IR::BailOnStackArgsOutOfActualsRange | IR::BailOutOnImplicitCallsPreOp));
+                //We are sure at this point, that we will not have any implicit calls as we don't have any call to helper.
+                instr->SetBailOutKind(IR::BailOnStackArgsOutOfActualsRange);
+            }
+        }
+        else if (instr->HasBailOutInfo() && (instr->GetBailOutKind() & IR::BailOnStackArgsOutOfActualsRange))
```

Comment:
oh sure. Done

---
In reply to: [68112935](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r68112935) [](ancestors = 68112935)